### PR TITLE
Windows app menu screen reader fixes

### PR DESCRIPTION
**Pull request by tidy-dev**
Closes [github/accessibility-audits#3250](https://github.com/github/accessibility-audits/issues/3250)

## Description
1) This adds the `tabindex="-1"` to the dropdown menu items so they are announced screen readers.
2) This adds the `role="menuitem"` to the menu bar buttons so they are announced by screen readers when using `alt` and left and right arrows. 
3) This focuses the first menu item when "Enter" or "Space" is used to open the menu.
4) Removes focus of menu pane that steals focus from first focused menu item. 
-  This last bullet may need closer looking at, maybe an additional `if` condition that focuses only if there is not an menu item selected.. which I am not sure when is.. but if that case were to be, then this would ensure the user could navigate the menu items with arrow keys. But, before I investigate further, I want to verify with accessibility team that what this is doing now is expected behavior. 

### Screenshots

https://user-images.githubusercontent.com/75402236/225020390-dcd52f2c-4f57-4adf-bb57-87fae2b44d8d.mp4

## Release notes
Notes: [Fixed] On Windows, app level menu bar and menu items are announced by screen readers.

### CODE REVIEW

File: app/src/ui/app-menu/app-menu.tsx [hunk 1]
```diff
@@ -333,7 +333,6 @@ export class AppMenu extends React.Component<IAppMenuProps, {}> {
     if (this.focusPane >= 0) {
       const pane = this.paneRefs[this.focusPane]
       if (pane) {
-        pane.focus()
```
Comment (tidy-dev) [inline app/src/ui/app-menu/app-menu.tsx:336]: @niik Do you remember context around this pane focusing? Currently, it is problematic and removing fixed my issue of stealing focus from the first selected menu item and therefore not announcing the first selected menu item.. but concerned it may break another scenario I am not testing. 

The problem I did encounter by removing this was that if a user hits `alt` and then opens a menu with `enter` , we did not select the first item and therefore we would need this to focus the pane so that up/down arrows would navigate it. I addressed this by doing what I think is the expectation (for windows) and focusing the first item on menu open. (Confirming with accessibility)

**Comment by tidy-dev** [comment]
Feedback on this PR from James Scholes:
> Okay, I tried the test build.  The menu items are now wreading out as expected, which is great.  Some comments about potential improvements:
1. You had indicated that the individual menu containers were being given a name with aria-label, and I suggested aria-labelledby instead pointing to the ID of the corresponding menu bar item.  Neither seems to be in place in this test build.  While NVDA won't report the menu name when it opens, other screen readers (e.g. JAWS) will. And so all menus must have a name.
2. The menu bar items aren't indicated as having an associated pop-up menu, although I could swear they used to be in the beta build.  Either way, they need aria-haspopup="menu".
3. The entire menu bar can be accessed by a screen reader user at all times, which isn't standard behaviour in Windows apps.  It should be hidden for all users unless a user actually tries to reach it with mouse or keyboard.
4. Once focus is in the menu bar, Tab and Shift+Tab move through the menu bar items similar to how Left/Right Arrow do.  This is also non-standard on Windows; pressing Tab from a menu bar should hide it and place focus where it was before.
Hope these are helpful!


I have implemented 1 and 2 on this PR. I think 3 and 4 are fallouts of this not being a standard windows menu but a custom menu, and are not accessibility violations (if anything they just make the feature useable in more ways tho maybe unexpected -> if we get additional feedback that this should change we can followup.) Since it is reading with screen readers as expected, I think the focus logic is not needed now that on access key we select the first item so I have removed it. <- happy to change that if I am missing something.

File: app/src/ui/app-menu/app-menu.tsx (hunk 1 above)
Comment (tidy-dev) [reply app/src/ui/app-menu/app-menu.tsx:336]: @niik Continued this conversation in [added comment](https://github.com/desktop/desktop/pull/16315#issuecomment-1472318476).

File: app/src/ui/lib/aria-types.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+export type AriaHasPopupType =
+  | boolean
+  | AriaHasPopup.Dialog
+  | AriaHasPopup.Menu
+  | AriaHasPopup.Listbox
+  | AriaHasPopup.Tree
+  | AriaHasPopup.Grid
+  | undefined
```
Comment (sergiou87) [inline app/src/ui/lib/aria-types.ts]: Another way to express this could be…
```suggestion
import { HTMLAttributes } from 'react'

export type AriaHasPopupType = HTMLAttributes<any>['aria-haspopup']
```

**Review by sergiou87 — COMMENTED** [review]
Code makes sense to me, I just found something that can be simplified if you want. Not a fan of the `any` I had to add, but I don't know another way of doing that 😆 

I can test it next week!

File: app/src/ui/lib/aria-types.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { HTMLAttributes } from 'react'
+
+export type AriaHasPopupType = HTMLAttributes<
+  string | boolean | undefined
```
Comment (sergiou87) [inline app/src/ui/lib/aria-types.ts]: While it seems to be irrelevant, I don't think this is correct 🤔 AFAIK `HTMLAttributes` expects the type to be an Element (if you dive into `DomAttributes` you will see that).

Anyway, just a very minor thing.

Comment (tidy-dev) [reply app/src/ui/lib/aria-types.ts]: K. I went googling around for examples of using this. It seems like `HTMLAttributes<HTMLElement>` might make sense. What do you think?

**Review by sergiou87 — APPROVED** [review]
For some reason I have trouble making it work the first time I press <kbd>Alt</kbd> and trying to move around with my keyboard:

https://user-images.githubusercontent.com/1083228/226945568-a0c46cf7-3371-499c-a321-9e65980e1cde.mov

Once I opened one and press <kbd>Escape</kbd>, it works as expected. Not sure if you want to address this in this PR or prefer another one, so I'm gonna approve this in case you want to go ahead and fix that later.

Unrelated to this PR, but I noticed that extra padding menu panes have on the right side, also present in `development` 🤯

**Comment by tidy-dev** [comment]
Yeah. I will look into this in a separate PR. ... and I know exactly why the extra padding is there.. easy fix for silly mistake 🤦 (overflow-y should be auto not scroll)

**Comment by tidy-dev** [comment]
> For some reason I have trouble making it work the first time I press Alt and trying to move around with my keyboard:

@sergiou87 
I have not been able to reproduce this... Maybe there is some additional step.
I am:
1. Opening beta
2. Hit alt. I can:
    - left/right navigate to any menu bar item and they are announced
    - I can hit down arrow, space, or enter to open a menu and have first item focused
3. I can hit alt+access key and it opens the menu with first item focused.
